src/main/java/net/minecraftforge/client/model/animation/Animation.java

```diff
@@ -0,0 +1,186 @@
+package net.minecraftforge.client.model.animation;
+
+import net.minecraft.util.ResourceLocation;
+import net.minecraft.world.World;
+import net.minecraftforge.client.model.IModelState;
+import net.minecraftforge.client.model.animation.ModelBlockAnimation.MBJointInfo;
+import net.minecraftforge.common.property.IUnlistedProperty;
+
+import com.google.common.base.Optional;
+import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
+import com.google.common.collect.ImmutableTable;
+
+public class Animation
+{
+    public static final IUnlistedProperty<IModelState> AnimationProperty = new IUnlistedProperty<IModelState>()
+    {
+        public String getName() { return "forge_animation"; }
+        public boolean isValid(IModelState state) { return true; }
+        public Class<IModelState> getType() { return IModelState.class; }
+        public String valueToString(IModelState state) { return state.toString(); }
+    };
+
+    public static float getWorldTime(World world)
+    {
+        return getWorldTime(world, 0);
+    }
+
+    public static float getWorldTime(World world, float tickProgress)
+    {
+        return (world.getTotalWorldTime() + tickProgress) / 20;
+    }
+
+    private static final AnimationStateMachine missing = new AnimationStateMachine(
```

Comment:
Hmm, there's Clips.IdentityClip that does nothing; state machine with 1 state being that clip might be useful. Main purpose of the missing AFSM is to signal the error from the loading code, but still provide some object, and it's not meant to be the identity object.